### PR TITLE
MWPW-144683 Article Header Unit Test

### PR DESCRIPTION
**Pull request by meganthecoder**
* Add unit test for the article header and add mocks
* Functionality for original PR (https://github.com/adobecom/bacom-blog/pull/37) was no longer needed but wanted to keep the unit tests

Related to: [MWPW-144683](https://jira.corp.adobe.com/browse/MWPW-144683)

**Test URLs:**
N/A

### CODE REVIEW

File: test/scripts/mocks/blocks/article-feed/article-helpers.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+}
+
+export function getLinkForTopic() {
+  return '#';
+}
+
+export function getTaxonomyModule() {
+  return true;
+}
```
Comment (meganthecoder) [inline test/scripts/mocks/blocks/article-feed/article-helpers.js:11]: To explain: 

These are here as mocks for the external milo scripts, as called out as best practice by the community. Before, I've tried to use import maps like we do in milo tests, but they don't work with template strings in dynamic imports. Here, I'm mocking the external scripts by setting milo libs to pull from the localhost mocks folder.

**Review by Brandon32 — APPROVED** [review]

File: test/scripts/mocks/blocks/article-feed/article-helpers.js (hunk 1 above)
Comment (meganthecoder) [reply test/scripts/mocks/blocks/article-feed/article-helpers.js:11]: @Brandon32 Let me know if you have further questions

**Comment by JasonHowellSlavin** [comment]
Not blocking, but I do wonder if this unit test doesn't belong in Milo itself? I understand we are testing the implementation that occurs in our utils.js, but wondering if we could skip some of the mocks by adding the test in milo? Or even what about the test in Milo was inadequate to catch our use case?

**Review by JasonHowellSlavin — APPROVED** [review]

**Comment by meganthecoder** [comment]
> Not blocking, but I do wonder if this unit test doesn't belong in Milo itself? I understand we are testing the implementation that occurs in our utils.js, but wondering if we could skip some of the mocks by adding the test in milo? Or even what about the test in Milo was inadequate to catch our use case?

I don't think that tests in milo would adequately cover our use case. The article header block in milo requires a very specific input or else it errors out and the block doesn't build correctly. That's essentially what we're testing here: Does the code in bacom blog utils provide correctly formatted data for the article header block to use?

Edit: It's possible that we could avoid mocking milo libs by rewriting `buildArticleHeader` but I don't see why we should. What's the objection to the way I'm mocking? It came from a suggestion by Okan.

**Comment by JasonHowellSlavin** [comment]
> > Not blocking, but I do wonder if this unit test doesn't belong in Milo itself? I understand we are testing the implementation that occurs in our utils.js, but wondering if we could skip some of the mocks by adding the test in milo? Or even what about the test in Milo was inadequate to catch our use case?
> 
> I don't think that tests in milo would adequately cover our use case. The article header block in milo requires a very specific input or else it errors out and the block doesn't build correctly. That's essentially what we're testing here: Does the code in bacom blog utils provide correctly formatted data for the article header block to use?
> 
> Edit: It's possible that we could avoid mocking milo libs by rewriting `buildArticleHeader` but I don't see why we should. What's the objection to the way I'm mocking? It came from a suggestion by Okan.

No objection to the way it's done. Just finding the test interesting, and what it necessitates interesting. "Does the code in bacom blog utils provide correctly formatted data for the article header block to use" is not something I think we've tested for in other consumer repos, and it exposes that the `article-header` may have been built incorrectly from the start according to current Milo practices. Nothing to be done in this PR, but food for thought nonetheless.

**Pull request merged**